server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java

```diff
@@ -226,6 +227,11 @@ public void onResponse(final Collection<ActionResponse> responses) {
 
             @Override
             public void onFailure(final Exception e) {
+                // Temporary logging to help debug https://github.com/elastic/elasticsearch/issues/45652
+                // TODO: remove this when we understand why _cat/indices sometimes returns a 404
+                if (e instanceof IndexNotFoundException) {
+                    logger.info("_cat/indices returning index_not_found_exception", e);
```

Comment:
I don't think we should add logging at INFO just for debugging tests. I think it could be DEBUG and the test logging level set to the appropriate `@TestLogging` annotation.

Reply:
We want the stack trace showing where the exception came from to be logged server side in the external test cluster, so I set the logger to debug in the test cluster settings rather than use `@TestLogging` (which would change the client side logger).

Reply:
Thanks!